python/ray/dag/compiled_dag_node.py

```diff
@@ -1111,6 +1125,14 @@ def _add_edge(
                 # Add an edge from the writer to the reader.
                 _add_edge(graph, idx, downstream_idx)
                 if task.dag_node.type_hint.requires_nccl():
+                    if _is_same_actor(idx, downstream_idx):
+                        logger.error(
+                            "Detect a deadlock caused by using NCCL channels to "
```

Comment:
Can you improve logs a little bit? it should show 1. actor id. 2 what's the name of tasks that run in the same actor

Reply:
Updated https://github.com/ray-project/ray/pull/46673/commits/cd72d64c74b6b39c311084b496b021c3ab99b737